tests/test_models_encodings.py

```diff
@@ -454,3 +456,46 @@ def test_ghz_circuit(backend, nqubits, density_matrix):
             target = backend.np.outer(target, backend.np.conj(target.T))
 
         backend.assert_allclose(state, target)
+
+
+# Helper to create a symmetric adjacency matrix
+def create_symmetric_adj(size=3):
+    G = nx.generators.cycle_graph(size)
+    return nx.adjacency_matrix(G).todense()
+
+
+# A non-numpy array input
+non_numpy_input = [[1, 2], [2, 1]]
+
+
+@pytest.mark.parametrize(
+    "matrix_input, expected_error",
+    [
+        # Valid symmetric matrices (should NOT raise an error)
+        (create_symmetric_adj(5), None),
+        (np.array([[1, 2], [2, 1]]), None),  # Manually created symmetric
+        (np.identity(3), None),  # Identity matrix is symmetric
+        # Invalid inputs (should raise errors)
+        (non_numpy_input, TypeError),  # Not a numpy array
+        (None, TypeError),  # None input
+        (123, TypeError),  # Integer input
+        (
+            np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=float),
+            ValueError,
+        ),  # non-symmetric
+        (np.array([[0, 1, 0], [0, 0, 1], [1, 0, 0]]), ValueError),  # non-symmetric
+    ],
+)
+def test_check_symmetric(matrix_input, expected_error):
+    if expected_error is None:
+        # For valid inputs, no error should be raised
+        try:
+            graph_state(matrix_input)
+            # If a print statement was inside, it would execute here.
+            # For tests, just successfully completing without error is the pass condition.
+        except Exception as e:
+            pytest.fail(f"Expected no error but got {type(e).__name__}: {e}")
+    else:
+        # For invalid inputs, the specific error should be raised
+        with pytest.raises(expected_error) as excinfo:
```

Comment:
You could look at examples of tests in the module to see how to better write tests, for instance, using `pytest.raises` instead of a `try/except`. For instance,

https://github.com/qiboteam/qibo/blob/5fe998de11072407e12275819803e6fed49caf26/tests/test_models_encodings.py#L101